src/engine/Module.v3

```diff
@@ -113,12 +113,14 @@ class Decl {
 def NO_EX_HANDLERS: Array<ExHandlerEntry> = [];
 def NO_DEST_MAP: Array<int> = [];
 def NO_DESTS: Array<TargetHandlerDest> = [];
+def NO_PC_MAP = HashMap<int, Vector<int>>.new(int.!, int.==);
 class FuncHandlerInfo {
 	var ex_handlers = NO_EX_HANDLERS;
 	var suspend_handlers = NO_EX_HANDLERS;
 	var switch_handlers = NO_EX_HANDLERS;
 	var handler_dest_map: Array<int> = NO_DEST_MAP;
 	var handler_dests: Array<TargetHandlerDest> = NO_DESTS;
+	var handler_pc_index_map = NO_PC_MAP;
```

Comment:
I think we should also modify the `findHandler()` method to use this new datastructure.

Reply:
I think it is a good idea to enforce the sorted array on all types of handlers so that `findHandler()` can do a binary search on each of them. I'm thinking instead of letting each `xxx_handlers` hold a handler entry, it should hold a handler index into the array of all handlers, and `handler_pc_index_map` would just be the array of all handlers in sorted order by `pc`. Without this change I don't think `findHandler` can use `handler_pc_index_map` as it is indexed by the source `pc` of the handler, not its range. What do you think?